sechub-testframework/src/main/java/com/daimler/sechub/test/PojoTester.java

```diff
@@ -10,92 +10,136 @@
-				if (optional) {
-					mockedArgument=Optional.of(mockedArgument);
-				}
-				writeMethod.invoke(objectToTest, mockedArgument);
-
-				Object result = readmethod.invoke(objectToTest);
-
-				if (result != mockedArgument) {
-					fail("The getter/setter implementation of " + objectToTest.getClass() + " failed on property:"
-							+ descriptor.getName());
-				}
-
-			}
-
-		} catch (Exception e) {
-			e.printStackTrace();
-			fail("Bean introspection failed - internal error in framework. See output for details - message was:"
-					+ e.getMessage());
-		}
-	}
+    private PojoTester() {
+
+    }
+
+    public static <T> void testEqualsAndHashCodeCorrectImplemented(T objectA, T objectBequalToA, T objectCnotEqualToAOrB) {
+        testBothAreEqualAndHaveSameHashCode(objectA, objectBequalToA);
+
+        testBothAreNOTEqual(objectA, objectCnotEqualToAOrB);
+
+    }
+
+    public static <T> void testBothAreNOTEqual(T objectA, T objectCnotEqualToAOrB) {
+        assertNotNull(objectCnotEqualToAOrB);
+        assertFalse("objectA is equals to objectC but must NOT be -check implementation!", objectA.equals(objectCnotEqualToAOrB));
+        assertFalse("objectB is equals to objectC but must NOT be -check implementation!", objectCnotEqualToAOrB.equals(objectA));
+    }
+
+    public static <T> void testBothAreEqualAndHaveSameHashCode(T objectA, T objectBequalToA) {
+        assertNotNull(objectA);
+        assertNotNull(objectBequalToA);
+
+        assertTrue("objectA is not equals not objectB but must be -check implementation!", objectA.equals(objectBequalToA));
+        assertTrue("objectB is not equals not objectA but must be -check implementation!", objectBequalToA.equals(objectA));
+
+        /* hash code check */
+        assertEquals("objectA has not same hashcode as objectB but must be -check implementation!", objectA.hashCode(), objectBequalToA.hashCode());
+    }
+
+    public static void testSetterAndGetter(Object objectToTest) {
+        assertNotNull("Cannot test null objects!", objectToTest);
+
+        try {
+            BeanInfo beanInfo = Introspector.getBeanInfo(objectToTest.getClass());
+            PropertyDescriptor[] propertyDescriptors = beanInfo.getPropertyDescriptors();
+            for (PropertyDescriptor descriptor : propertyDescriptors) {
+                Method writeMethod = descriptor.getWriteMethod();
+                if (writeMethod == null) {
+                    continue;
+                }
+                Method readmethod = descriptor.getReadMethod();
+                if (readmethod == null) {
+                    continue;
+                }
+                Class<?> type = descriptor.getPropertyType();
+                boolean optional = type.equals(Optional.class);
+                if (optional) {
+                    /* ugly but necessary, to handle optinonal... */
+                    Type generic = readmethod.getGenericReturnType();
+                    ParameterizedType pm = (ParameterizedType) generic;
```

Comment:
What is `pm` please provide a better name. This code is already very abstract.

Reply:
changed to parameterizedType: done

Reply:
I changed the class, separated into different methods, added comments, javadoc etc. to make it less abstract and easier to read. Hope this helps and makes it easier to understand.